lilypad/prompts.py

```diff
@@ -30,26 +30,47 @@
 
 if TYPE_CHECKING:
     try:
-        from mirascope.core.openai import OpenAICallParams
-        from openai.types.chat import (
-            ChatCompletionMessageParam,  # pyright: ignore [reportAssignmentType]
-        )
+        from anthropic.types import MessageParam
```

Comment:
Why did you change the type checking import format. Just add 
```
from mirascope.core.groq import GroqCallParams
from groq.types.chat import ChatCompletionMessageParam as GroqMessageParam
...
```
as needed rather than separating mirascope and the provider